src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReferencesPattern.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Raven.Server.Documents.Indexes.MapReduce.OutputToCollection
+{
+    public class OutputReferencesPattern
+    {
+        private static readonly Regex FieldsRegex = new Regex(@"\{([^\:}]*)\:?([^}]*)\}", RegexOptions.Compiled);
+
+        private readonly DocumentIdBuilder _builder;
+
+        public OutputReferencesPattern(string pattern)
+        {
+            var matches = FieldsRegex.Matches(pattern);
+
+            if (matches.Count == 0)
+                throw new InvalidOperationException("Provided pattern is not supported: " + pattern);
+            
+            var fieldToFormatPosition = new Dictionary<string, int>(matches.Count);
+            int numberOfFields = 0;
+
+            var formattedPattern = FieldsRegex.Replace(pattern, StringFormatEvaluator);
+
+            _builder = new DocumentIdBuilder(formattedPattern, fieldToFormatPosition);
+
+            string StringFormatEvaluator(Match m)
+            {
+                var groups = m.Groups;
+
+                try
+                {
+                    if (groups.Count == 2)
+                    {
+                        // {OrderedAt}
+
+                        return $"{{{numberOfFields}}}";
+                    }
+
+                    if (groups.Count == 3)
+                    {
+                        // {OrderedAt:yyyy-MM-dd}
+
+                        return $"{{{numberOfFields}:{groups[2]}}}";
```

Comment:
what happens when the pattern we have here is not valid?

Reply:
it will be handled as regular indexing error